src/iosource/Manager.cc

```diff
@@ -59,124 +109,411 @@ IOSource* Manager::FindSoonest(double* ts)
 			break;
 			}
 
-	// Ideally, we would always call select on the fds to see which
-	// are ready, and return the soonest. Unfortunately, that'd mean
-	// one select-call per packet, which we can't afford in high-volume
-	// environments.  Thus, we call select only every SELECT_FREQUENCY
-	// call (or if all sources report that they are dry).
+	// If there aren't any sources and exit_only_after_terminate is false, just
+	// return an empty set of sources. We want the main loop to end.
+	if ( Size() == 0 && ( ! BifConst::exit_only_after_terminate || terminating ) )
+		return ready;
 
-	++call_count;
+	double timeout = -1;
+	IOSource* timeout_src = nullptr;
+	timer_expired = false;
 
-	IOSource* soonest_src = 0;
-	double soonest_ts = 1e20;
-	double soonest_local_network_time = 1e20;
-	bool all_idle = true;
+	PktSrc* pkt_src = pkt_srcs.empty() ? nullptr : pkt_srcs.front();
```

Comment:
We can only have one pkt src now, so need for list anymore. 

Though actually, I lost track, let me ask: what's the reason now that we cannot support more than one packet source anymore?

Reply:
In the existing code, we check every single packet source to see what the next thing is. That includes each of the packet sources, and the code will choose the individual packet source that has the next packet in the timeline. This means that if you have two packet files that are interleaved, it'll switch back and forth between them.

The new code doesn't have that capability anymore. It just checks each of the sources to get a timeout value for the poll call. That said, maybe it could do the old way? If it looped through the packet sources first, independent of the rest of the sources, it could store off which packet source is next and then return that if it's ready. That would work for packet files. I'm not sure how it would work for live sources (is that a thing people do?).

Reply:
Got it. In the interest of not special-casing packet sources more than we need to, seems fine to leave as is then, given that there wasn't any push back on removing the capability.